cmd/podman/system/service_abi.go

```diff
@@ -24,16 +24,28 @@ func restService(opts entities.ServiceOptions, flags *pflag.FlagSet, cfg *entiti
 	)
 
 	if opts.URI != "" {
-		fields := strings.Split(opts.URI, ":")
-		if len(fields) == 1 {
-			return errors.Errorf("%s is an invalid socket destination", opts.URI)
+		if os.Getenv("LISTEN_FDS") != "" {
+			// If it is activated by systemd, use the first LISTEN_FD (3)
+			// instead of opening the socket file.
+			f := os.NewFile(uintptr(3), "podman.sock")
```

Comment:
Should this be hard coded to 3 or should we use the value of LISTEN_FDS?

Reply:
LISTEN_FDS says how many fds there are but we always need the first one.

It is the first FD after 0, 1, 2 so it can be hardcoded to 3